cloudigrade/util/tests/redhatcloud/test_sources.py

```diff
@@ -123,63 +133,151 @@ def test_get_sources_cloudigrade_application_type_fail(self, mock_get):
-
-        mock_patch.assert_called()
-
-    @patch("requests.patch")
-    def test_notify_sources_application_availability_other_status(self, mock_patch):
-        """Test unknown status for patching sources."""
+        mock_kafka_producer.assert_not_called()
+
+    @patch("util.redhatcloud.sources.KafkaProducer")
+    def test_notify_sources_application_availability_buffer_error(
+        self,
+        mock_kafka_producer,
+    ):
+        """Test notify source application availability handling BufferError."""
         application_id = _faker.pyint()
         availability_status = "available"
 
-        mock_patch.return_value.status_code = http.HTTPStatus.INTERNAL_SERVER_ERROR
-
-        with self.assertRaises(SourcesAPINotOkStatus):
-            sources.notify_application_availability(
-                self.account_number,
-                application_id,
-                availability_status=availability_status,
+        sources_kafka_config = {
+            "bootstrap.servers": f"{self.listener_server}:{self.listener_port}"
+        }
+
+        kafka_producer = mock_kafka_producer(sources_kafka_config)
+        kafka_producer.produce.side_effect = BufferError("bad error")
+
+        payload = {
+            "resource_type": self.sources_resource_type,
+            "resource_id": application_id,
+            "status": availability_status,
+            "error": "",
+        }
+
+        with override_settings(
+            LISTENER_SERVER=self.listener_server,
+            LISTENER_PORT=self.listener_port,
+            SOURCES_STATUS_TOPIC=self.sources_kafka_topic,
+            SOURCES_RESOURCE_TYPE=self.sources_resource_type,
+            SOURCES_AVAILABILITY_EVENT_TYPE=self.sources_availability_event_type,
+            SOURCES_ENABLE_DATA_MANAGEMENT_FROM_KAFKA=True,
+        ):
+            with self.assertRaises(KafkaProducerException):
+                sources.notify_application_availability(
+                    application_id,
+                    availability_status=availability_status,
+                )
+
+            kafka_producer.poll.assert_called_once_with(0)
+            kafka_producer.produce.assert_called_with(
+                topic=self.sources_kafka_topic,
+                value=json.dumps(payload),
+                headers={"event_type": self.sources_availability_event_type},
             )
+            kafka_producer.flush.assert_not_called()
+
+    @patch("util.redhatcloud.sources.KafkaProducer")
+    def test_notify_sources_application_availability_kafka_exception(
+        self,
+        mock_kafka_producer,
+    ):
+        """Test notify source application availability handling KafkaException."""
+        application_id = _faker.pyint()
+        availability_status = "available"
 
-        mock_patch.assert_called()
+        sources_kafka_config = {
+            "bootstrap.servers": f"{self.listener_server}:{self.listener_port}"
+        }
+
+        kafka_producer = mock_kafka_producer(sources_kafka_config)
```

Comment:
I see these same lines being repeated in several tests. I think they might be a good candidate for moving into `setUp` and defined as instance variables on `self` to keep the individual tests DRY.

Ditto for the `payload = {` definition a few lines below.